src/hermite_multidimensional.hpp

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019 Xanadu Quantum Technologies Inc.
+
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <stdafx.h>
+
+#include <Eigen/Eigenvalues>
+#include <Eigen/LU>
+
+typedef unsigned long long int ullint;
+
+
+ullint vec2index(std::vector<int> &pos, int resolution) {
+    int dim = pos.size();
+    ullint nextCoordinate = 0;
+
+    nextCoordinate = pos[0]-1;
+    for(int ii = 0; ii < dim-1; ii++) {
+        nextCoordinate = nextCoordinate*resolution + (pos[ii+1]-1);
+    }
+
+    return nextCoordinate;
+
+}
+
+std::vector<int> find_rep(int val, int base, int n) {
+    std::vector<int> x(n, 0);
+    int local_val = val;
+
+    x[0] = 1;
+
+    for (int i = 1; i < n; i++)
+        x[i] = x[i-1]*base;
+
+    std::vector<int> digits(n, 0);
+
+    for (int i = 0; i < n; i++) {
+        digits[i] = local_val/x[n-i-1];
+        local_val = local_val - digits[i] * x[n-i-1];
+    }
+
+    return digits;
+
+}
+
+
+long double sqrtfactorial(int nn)
+{
+    long double n = static_cast<long double>(nn);
+
+    if(n > 1)
+        return std::sqrt(n) * sqrtfactorial(n - 1);
+    else
+        return 1;
+}
+
+template <typename T>
+inline std::vector<T> renormalization(std::vector<T> tn, int nmodes, int res) {
+    std::vector<long double> invsqfacts(res, 0);
+    std::vector<int> digits(nmodes, 0);
+
+    ullint Hdim = pow(res, nmodes);
+
+    for (int i = 0; i < res; i++)
+        invsqfacts[i] = sqrtfactorial(i);
+
+    for (ullint i = 0; i < Hdim; i++) {
+        digits = find_rep(i, res, nmodes);
+        long double pref = 1;
+        for (int j = 0; j < nmodes; j++)
+            pref *= 1.0L/invsqfacts[digits[j]];
+        tn[i] = tn[i]*static_cast<double>(pref);
+    }
+
+    return tn;
+
+}
```

Comment:
Don't forget to add doxygen docstrings for the above functions! 🙂

Reply:
Done!